### ISSUE TITLE
introduce more module add parameters

### ISSUE THREAD

**Issue by propensive** [propensive/fury-old#1303]
It should be possible to add a module defining the following, all in the `add` command,
- sources
- dependencies
- binaries
- parameters
- environment variables
- resources
as they all have different parameter keys.

**Comment by odisseus** [comment]
I don't think that would be a useful addition to the interface. In my opinion, two short commands are more convenient than a single long one, as long as they are quick and reliable.

Moreover, such long command would mix together properties of the module object (such as its type and main class), and independent entities which have their own properties (such as binaries and dependencies). How would this command allow the user to set those properties?

**Comment by propensive** [comment]
It's something people have asked for, and equally people complain about having to type 3-4 commands just to create a module with the right settings. It's possible in general because most of those entities can be described with a single tab-completable string, even if we ignore some additional parameters such as the transitivity of dependencies.

**Comment by veyga** [comment]
Some command line tools give you step-wise command entries when the overall command is long. Rather than a single long command, it could be something like 
fury module add
-name? [required]:  (enter name)
-compiler? [default=javac]: (enter compiler)
-type? [default = library]: ...
-sources? ...
etc...
 Not sure how that would play with the existing parsing mechanisms, but from experience, as a user, I'd much rather work with that than trying to type out one long command, maven style.

**Comment by propensive** [comment]
I think having an interactive mode like this for commands would be good. I think that I'd like to have it enabled only with an `--interactive`/ `-I` flag, so that users can safely know that they can put `fury` commands in a script and they won't block on user input. So far, no `fury` commands are interactive. But we could create a separate issue to add them.

**Comment by odisseus** [comment]
If we ever have the interactive prompts, they should be enabled by default, otherwise the user might never learn that they exist. It would be nice if they could be explicitly disabled for the purpose of writing scripts, but Bash and Fish shells already provide a way to [check](https://unix.stackexchange.com/a/26782) whether the command is executed in an interactive shell.

**Comment by odisseus** [comment]
Another possible option is creating a separate command that would guide the user through every decision needed to set up a layer from scratch, such as creating projects and modules, selecting the compiler, importing other layers etc.
